pkg/pipelinescheduler/config.go

```diff
@@ -0,0 +1,388 @@
+	// TrustedOrg is the org whose members' PRs will be automatically built
+	// for PRs to the above repos. The default is the PR's org.
+	TrustedOrg *string `json:"trustedOrg,omitempty"`
+	// JoinOrgURL is a link that redirects users to a location where they
+	// should be able to read more about joining the organization in order
+	// to become trusted members. Defaults to the Github link of TrustedOrg.
+	JoinOrgURL *string `json:"joinOrgUrl,omitempty"`
+	// OnlyOrgMembers requires PRs and/or /ok-to-test comments to come from org members.
+	// By default, trigger also include repo collaborators.
+	OnlyOrgMembers *bool `json:"onlyOrgMembers,omitempty"`
+	// IgnoreOkToTest makes trigger ignore /ok-to-test comments.
+	// This is a security mitigation to only allow testing from trusted users.
+	IgnoreOkToTest *bool `json:"ignoreOkToTest,omitempty"`
+}
+
+// Postsubmits is a list of Postsubmit job configurations that can optionally completely replace the Postsubmit job
+// configurations in the parent scheduler
+type Postsubmits struct {
+	// Items are the post submit configurations
+	Items []*Postsubmit
+	// Replace the existing items
+	Replace bool
+}
+
+// Postsubmit runs on push events.
+type Postsubmit struct {
+	*JobBase
+
+	*RegexpChangeMatcher
+
+	*Brancher
+
+	// Context is the name of the GitHub status context for the job.
+	Context *string `json:"context"`
+
+	// Report will comment and set status on GitHub.
+	Report *bool `json:"report,omitempty"`
+}
+
+// JobBase contains attributes common to all job types
+type JobBase struct {
+	// The name of the job. Must match regex [A-Za-z0-9-._]+
+	// e.g. pull-test-infra-bazel-build
+	Name *string `json:"name"`
+	// ReplaceableMapOfStringString are added to jobs and pods created for this job.
+	Labels *ReplaceableMapOfStringString `json:"labels,omitempty"`
+	// MaximumConcurrency of this job, 0 implies no limit.
+	MaxConcurrency *int `json:"maxConcurrency,omitempty"`
+	// Agent that will take care of running this job.
+	Agent *string `json:"agent"`
+	// Cluster is the alias of the cluster to run this job in.
+	// (Default: kube.DefaultClusterAlias)
+	Cluster *string `json:"cluster,omitempty"`
+	// Namespace is the namespace in which pods schedule.
+	//   empty: results in scheduler.DefaultNamespace
+	Namespace *string `json:"namespace,omitempty"`
+}
+
+// ReplaceableMapOfStringString is a map of strings that can optionally completely replace the map of strings in the
+// parent scheduler
+type ReplaceableMapOfStringString struct {
+	Items map[string]string
+	// Replace the existing items
+	Replace bool
+}
+
+// Presubmits is a list of Presubmit job configurations that can optionally completely replace the Presubmit job
+// configurations in the parent scheduler
+type Presubmits struct {
+	// Items are the Presubmit configurtations
+	Items []*Presubmit
+	// Replace the existing items
+	Replace bool
+}
+
+// Presubmit defines a job configuration for pull requests
+type Presubmit struct {
```

Comment:
Is this copied from `prow/jobs.go`? It will be challenging to keep it in sync. I am wondering if we can use directly the prow package here.

Reply:
It's similar but not the same. We can't reuse as (a) the prow config is a mess and (b) the structure is different.